### DISCUSSION TITLE
Why people can't find their public SSH keys

### DISCUSSION THREAD

**Discussion by AnxiousLeopard** [community/community#27738] [Other Feature Feedback, Questions, & Ideas] [answered]
I just had the same issue as many people who created their shiny new SSH key in GIT bash following the instructions and realised that there is a Microsoft Publisher document appearing without the file extension when the file is marked as ".pub"
If you right-click this and open it in notepad you will have your public key. I have shared this on recent discussion threads

This will keep happening to everyone on a Windows OS Machine with MS Office, unless the file extension is changed, or the procedures are rewritten. It took a lot of head scratching and a lot of wasted time for a reasonably intelligent, experienced person whose first language is English to figure this out. I know others will not find it so easy. 

Is there anything that can be done to change the file extension on public SSH keys in future that won't break the way things work? 
Otherwise, this will be an ongoing issue.

**Comment by airtower-luna** [comment, marked as answer]
> Is there anything that can be done to change the file extension on public SSH keys in future that won't break the way things work?

No, that would break backwards compatibility with approximately every SSH implementation. The correct solution is to disable file extension hiding. I don't use Windows myself, but I remember from when I last had to that there was an option for that. I agree that mentioning that in the documentation would be helpful. :slightly_smiling_face:

**Reply by AnxiousLeopard** [reply]
Thanks for pointing out being able to disable file extension hiding. 
I have turned this off for now. It wasn't an obvious setting in Windows 10 Home, but a search in the settings for "file extension" brought me to what I needed. 
Sharing this specifically for anyone else who falls down a similar rabbit hole.

**Reply by WHYME31** [reply]
It still opens windows publisher for me if I tried to open it with the cmd line 
![image](https://github.com/community/community/assets/92877366/3db0c50e-264c-45db-8aa2-4c6e5d1b5df2)
